### ISSUE TITLE
Should we disable asp0014 when using `WebApplication.CreateEmptyBuilder()`

### ISSUE THREAD

**Issue by WeihanLi** [dotnet/aspnetcore#50732]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Since the empty WebAplicationBuilder would not register routing, maybe we could disable the `ASP0014` using top-level route registrations diagnostic?

We may use the code like below, currently we would get `ASP0014` warning
![image](https://github.com/dotnet/aspnetcore/assets/7604648/43c77567-5609-4531-aa98-95d9ee910e53)

``` c#
var builder = WebApplication.CreateEmptyBuilder(new WebApplicationOptions());
builder.Services.AddRoutingCore();
builder.WebHost.UseKestrelCore();
builder.Logging.AddConsole();

var app = builder.Build();

app.UseRouting();
app.UseEndpoints(endpoints => 
{
    endpoints.MapGet("/", () => "Hello World");
});

app.Run();
```

- https://github.com/dotnet/aspnetcore/pull/49246
- https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0014

### Describe the solution you'd like

Ignore it when using the empty `WebApplicationBuilder` since no default global routing

**Comment by captainsafia** [comment]
Thanks for the bug report, @WeihanLi! Yes, we should avoid emitting this diagnostic for web apps not configured with the routing middleware.

Let me see how easy this is to fix on the analyzer side...

**Comment by david-acker** [comment]
I just took a quick look, but it seems like this would be relatively straight forward. Right now we're emitting the diagnostic for any method on `WebApplication`.

Maybe we could pass an array of excluded method symbols  to the `IsDisallowedMethod` method in `WebApplicationBuilderAnalyzer`? That way we could check that the method we're analyzing isn't one we want to exclude, after we check the method's containing type [here](https://github.com/dotnet/aspnetcore/blob/0126698158fcdf06596e36899424403d0a45847f/src/Framework/AspNetCoreAnalyzers/src/Analyzers/WebApplicationBuilder/WebApplicationBuilderAnalyzer.cs#L352C17-L352C94).

**Comment by captainsafia** [comment]
> Seemed we're emitting the ASP0014 diagnostic for UseEndpoints method only, we may need to distinguish whether the WebApplication is an empty web application, if it is, we may need to exclude it, while it seems there's no property or something indicates if it's empty for now

Yep! The challenge here is that we want to continue emitting the warning about `UseEndpoints` for non-empty `WebApplicationBuilder`s. There's not a straightforward way for us to determine if a given builder is empty or not unless we do some flow analysis to determine if it was constructed via `CreateEmptyBuilder.`

**Comment by captainsafia** [comment]
Backlogging this for now. Was hopping I'd be able to tackle this for 9.0-preview1 but that turned out to not be the case.

PRs welcome on this though!
